README.md

```diff
@@ -97,6 +97,12 @@ end
     Z = promote_type(X,Y) # calculates Z from X and Y
     fun5(X,Y) -> Z
 end
+
+# using parametric trait. Note the nested curly
+@traitdef SemiFunctor{X{Y}} begin
+    fmap( Function, X{Y} } -> Any
+end
+
```

Comment:
Yes I thought about that, but for staging to work, I think we would need access to the input/output type of functions during the staging phase. Here's a recent comment I made on this topic: https://github.com/JuliaLang/julia/pull/7474#issuecomment-69329613